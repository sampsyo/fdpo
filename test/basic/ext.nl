# ARGS: x=9
in x: 4;
out s: 8;
out z: 8;
out e: 2;
s = sext[4, 8](x);
z = zext[4, 8](x);
e = slice[4, 2, 3](x);
