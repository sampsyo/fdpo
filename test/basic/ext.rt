in x: 4;
out s: 8;
out z: 8;
s = sext[4, 8](x);
z = zext[4, 8](x);
