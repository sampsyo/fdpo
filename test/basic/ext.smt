(let ((.def_0 ((_ zero_extend 4) x))) (let ((.def_1 (= z .def_0))) (let ((.def_2 ((_ sign_extend 4) x))) (let ((.def_3 (= s .def_2))) (let ((.def_4 (and .def_3 .def_1))) .def_4)))))
