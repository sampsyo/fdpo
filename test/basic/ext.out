s = 249
z = 9
